Fix #2752 - Allow domain recipients for address rewrite

PseudoResonance opened the pull request:
<!-- _Please make sure to review and check all of these items, otherwise we might refuse your PR:_ -->

## Contribution Guidelines

* [x] I've read the [contribution guidelines](https://github.com/mailcow/mailcow-dockerized/blob/master/CONTRIBUTING.md) and wholeheartedly agree them

<!-- _NOTE: this tickbox is needed to fullfil on order to get your PR reviewed._ -->

## What does this PR include?

### Short Description

Fix for #2752 to allow address rewriting to entire domains, instead of being limited to a single email.
I simply copied the old address validation code and replaced the new address validation so that they both share the same code.
<!-- Please write a short description, what your PR does here. -->

###  Affected Containers

<!-- Please list all affected Docker containers here, which you commited changes to -->
 - nginx
<!--

Please list them like this:

- container1
- container2
- container3
etc.

-->

## Did you run tests?

### What did you tested?

I inputted a number of different valid/invalid domain and email combinations to ensure the validation works as intended.
<!-- Please write shortly, what you've tested (which components etc.). -->

### What were the final results? (Awaited, got)

Works as described in #2752 and either exact emails or whole domains can be added as a new address.
<!-- Please write shortly, what your final tests results were. What did you awaited? Was the outcome the awaited one? -->

PseudoResonance committed:
Fix #2752: Domain recipient for address rewrite

(cherry picked from commit 40f6d691d8774d6f813153974f8fe462a8db9ab3)

milkmaker commented:
This pull request has been automatically marked as stale because it has not had recent activity. It will be closed if no further activity occurs.

PseudoResonance commented:
I rather dislike these stale bots...

FreddleSpl0it reviewed (changes requested):
Please also apply this logic to `$new_dest`
```
      if (substr($old_dest, 0, 1) == '@') {
        $old_dest = substr($old_dest, 1);
      }
```

After adding a domain as `$new_dest`, an `@` is prepended. Editing this entry will only work if the `@` is removed.

PseudoResonance committed:
Fix @ prefixing domain rewrite and update localization

PseudoResonance commented:
> After adding a domain as `$new_dest`, an `@` is prepended. Editing this entry will only work if the `@` is removed.

I think I implemented all of `old_dest`'s logic to `new_dest` as well. It now removes the prefixing `@` when editing it and properly handles it when included.

Additionally, I updated what localizations I could to indicate that an email or domain is valid. I simply used the old dest translation, and updated the new dest one to match. I checked them with Google translate and I believe I got all the grammar correct in the languages I could check, however I'm only fully confident about English, Japanese and Spanish. The rest I do not speak and just had to use the internet.

A few languages I couldn't confidently update though, as the old_dest translation is inaccurately not mentioning domains as being valid, or were missing the translation. For example, Ukrainian, Simplified Chinese, Taiwanese.

I can remove them though if you want them in another commit or something.
